devel/fortio/http.go

```diff
@@ -696,19 +705,41 @@ func EchoHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-// DynamicHTTPServer listens on an available port and return it.
-func DynamicHTTPServer() int {
+// DynamicHTTPServer listens on an available port, sets up an http or https
+// (when secure is true) server on it and returns the listening port.
+func DynamicHTTPServer(secure bool) int {
 	listener, err := net.Listen("tcp", ":0")
 	if err != nil {
 		Fatalf("Unable to listen to dynamic port: %v", err)
 	}
 	port := listener.Addr().(*net.TCPAddr).Port
 	Infof("Using port: %d", port)
-	go func(port int) {
-		if err := http.Serve(listener, nil); err != nil {
-			Fatalf("Unable to serve on %d: %v", port, err)
+	go func(secure bool, port int) {
```

Comment:
question: any reason not make this a separate method (or two)?

something like:
```
if secure {
    go runSecure(port, listener)
    return port
}
go runInsecure(port, listener)
return port
```

Reply:
it was going to be a 1 liner: `ServeTLS` except that's a 1.9 feature

Reply:
changed